integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsCompleteAvailableCondition.java

```diff
@@ -0,0 +1,922 @@
+
+    // verify the admin server service exists
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace1);
+
+    // verify the cluster server pods are up and running
+    logger.info("Checking managed server pods were ready");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPodNamePrefix + i, domainUid, domainNamespace1);
+    }
+
+    // verify the condition type Completed exists
+    checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE);
+    // verify the condition type Available exists
+    checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE);
+    // verify the condition Completed type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True");
+    // verify the condition Available type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "True");
+    // verify there is no status condition type Failed
+    verifyDomainStatusConditionTypeDoesNotExist(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_FAILED_TYPE);
+  }
+
+  /**
+   * Test domain status condition with new image.
+   * Verify all the servers pods are restarted.
+   * Verify the following conditions are generated:
+   * type: Completed, status: true
+   * type: Available, status: true
+   * Verify no Failed type condition generated.
+   */
+  @Order(10)
+  @Test
+  @DisplayName("Test domain status condition with new restartVersion")
+  void testCompleteAvailableConditionWithNewImage() {
+
+    // get the pod creation time stamps
+    LinkedHashMap<String, OffsetDateTime> pods = new LinkedHashMap<>();
+    // get the creation time of the admin server pod before patching
+    OffsetDateTime adminPodCreationTime = getPodCreationTime(domainNamespace1, adminServerPodName);
+    pods.put(adminServerPodName, adminPodCreationTime);
+    // get the creation time of the managed server pods before patching
+    for (int i = 1; i <= replicaCount; i++) {
+      pods.put(managedServerPodNamePrefix + i,
+          getPodCreationTime(domainNamespace1, managedServerPodNamePrefix + i));
+    }
+
+    Domain domain = assertDoesNotThrow(() -> getDomainCustomResource(domainUid, domainNamespace1));
+    //print out the original image name
+    String imageName = domain.getSpec().getImage();
+    logger.info("Currently the image name used for the domain is: {0}", imageName);
+
+    //change image name to imageUpdate
+    String imageTag = CommonTestUtils.getDateAndTimeStamp();
+    String imageUpdate = KIND_REPO != null ? KIND_REPO
+        + (WEBLOGIC_IMAGE_NAME + ":" + imageTag).substring(TestConstants.BASE_IMAGES_REPO.length() + 1)
+        : WEBLOGIC_IMAGE_NAME + ":" + imageTag;
```

Comment:
I guess you don't need to tinker with the REPO, The repo name is automatically set in the IMAGE_NAME. Something like this.

```
    String newImage = MII_BASIC_IMAGE_NAME + ":" + tag;
    dockerTag(miiImage, newImage);
    dockerLoginAndPushImageToRegistry(newImage);
```

Reply:
done